### ISSUE TITLE
Search fails for some deps

### ISSUE THREAD

**Issue by agzam** [babashka/neil#27]
Version 0.0.23

```
neil dep search "org.clojure/tools.cli"
----- Error --------------------------------------------------------------------
Type:     java.lang.NullPointerException
Location: /usr/local/bin/neil:337:11

----- Context ------------------------------------------------------------------
333:   (let [search-term (first (:cmds opts))
334:         url (str "https://clojars.org/search?format=json&q=" search-term)
335:         {search-results :results
336:          results-count :count} (curl-get-json url)]
337:     (when (zero? results-count)
               ^---
338:       (binding [*out* *err*]
339:         (println "Unable to find" search-term  "on Clojars.")
340:         (System/exit 1)))
341:     (doseq [search-result search-results]
342:       (println :lib (format  "%s/%s"

----- Stack trace --------------------------------------------------------------
clojure.core/zero?       - <built-in>
babashka.neil/dep-search - /usr/local/bin/neil:337:11
babashka.neil/dep-search - /usr/local/bin/neil:332:1
babashka.neil/dep        - /usr/local/bin/neil:414:16
babashka.neil/dep        - /usr/local/bin/neil:408:1
babashka.neil/-main      - /usr/local/bin/neil:420:13
babashka.neil/-main      - /usr/local/bin/neil:416:1
babashka.neil            - /usr/local/bin/neil:425:3
```

**Comment by borkdude** [comment]
This is a problem we should indeed fix. The problem is likely the slash in the name.

**Comment by bobisageek** [comment]
It seems like the slash does break search, even on the clojars website (searching for org.clojure/tools.cli without quotes returns `Could not search; please check your query syntax.`).

I think quoting the search term might be the way to go:
```shell
$ ./neil dep search '"org.clojure/tools.cli"'
:lib org.clojars.rmremizov/clj-argparse :version 0.1.1-SNAPSHOT
```
^
This lib has the term "org.clojure/tools.cli" in its description.

But it should be noted that searching like this won't return a namespace-qualified lib, because clojars appears to store the namespace and name in separate fields (namely group-id and artifact-id).
```shell
$ ./neil dep search '"babashka.nrepl"'
:lib babashka/babashka.nrepl :version 0.0.6
$ ./neil dep search '"babashka/babashka.nrepl"'
Unable to find "babashka/babashka.nrepl" on Clojars.
```

Presumably the `versions` command would be more aimed at supplying an exact lib name.

The change of quoting the search term would _technically_ break the capability to do lucene-type searches just like on the clojars website:
```shell
$ ./neil dep search 'babashka+AND+nrepl'
:lib babashka/babashka.nrepl :version 0.0.6
$ ./neil dep search 'babashka+OR+nrepl'
:lib nrepl/nrepl :version 0.9.0
:lib cider/cider-nrepl :version 0.28.3
...
```
but the README says things might break, and it seems unlikely to me that this functionality would be in heavy use.


Tangentially, I'd like to propose that the search term also be URL-encoded to allow for spaces in search strings:
```shell
$ ./neil dep search '"babashka pods"'
----- Error --------------------------------------------------------------------
Type:     com.fasterxml.jackson.core.JsonParseException
...(details elided)
```

vs

```shell
$ ./neil dep search '"babashka+pods"'
:lib babashka/babashka.pods :version 0.1.0
```

TL;DR - Quoting and URL-encoding the search string would eliminate a large swath of the error responses that lead to exceptions and facilitate searching for multi-word strings, but namespace-qualified libs won't be found.

I've got the start of a PR for this (need to do some cleanup and add some tests), but I wanted to provide the context for feedback.

**Comment by borkdude** [comment]
Thanks! Makes a lot of sense.

**Comment by borkdude** [comment]
Published as v0.0.24, thanks @bobisageek
